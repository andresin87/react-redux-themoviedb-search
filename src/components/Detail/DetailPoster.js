import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { selectorDetail } from '../../reducers';

class DetailPoster extends PureComponent {
  render() {
    const {
      detail: {
        movie: { title, poster_path },
      },
    } = this.props;
    if (!poster_path) return null;
    return (
      <img
        alt={title}
        style={{ width: 500 }}
        src={`https://image.tmdb.org/t/p/w500${poster_path}`}
      />
    );
  }
}

DetailPoster.propTypes = {
  detail: PropTypes.object.isRequired,
};

const mapStateToProps = state => {
  return { detail: selectorDetail(state) };
};

export default connect(mapStateToProps)(DetailPoster);
