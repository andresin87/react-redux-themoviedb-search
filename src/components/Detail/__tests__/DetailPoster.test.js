import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import DetailPoster from '../DetailPoster';
import store from '../../../config/mockStore';

describe('DetailPoster component', () => {
  const component = (
    <Provider store={store}>
      <DetailPoster />
    </Provider>
  );

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
