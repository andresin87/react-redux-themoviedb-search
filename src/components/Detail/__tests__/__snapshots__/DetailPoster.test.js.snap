// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPoster component Snapshot matchs 1`] = `
<img
  alt="Fight Club"
  src="https://image.tmdb.org/t/p/w500/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg"
  style={
    Object {
      "width": 500,
    }
  }
/>
`;
